src/main/app-updater.ts

```diff
@@ -1,19 +1,144 @@
+        autoUpdater.quitAndInstall();
+
+        resolve();
+      })
+      .on(yesLaterChannel, async () => {
+        cleanupChannels();
+
+        await autoUpdater.downloadUpdate();
+        autoUpdater.autoInstallOnAppQuit = true;
 
-  public start() {
-    setInterval(() => {
-      autoUpdater.checkForUpdatesAndNotify()
-    }, this.updateInterval)
+        resolve();
+      })
+      .on(noChannel, () => {
+        cleanupChannels();
+        resolve();
+      });
 
-    return autoUpdater.checkForUpdatesAndNotify()
+    windowManager.mainView.webContents.send(NotificationChannelAdd, {
+      args: [{
+        title,
+        body,
+        status: "info",
+        buttons: [
+          {
+            label: "Yes, now",
+            backchannel: yesNowChannel,
+            action: true,
+          },
+          {
+            label: "Yes, on quit",
+            backchannel: yesLaterChannel,
+            action: true,
+          },
+          {
+            label: "No",
+            backchannel: noChannel,
+            secondary: true
+          }
+        ],
+        closeChannel: noChannel,
+      }]
+    });
+  });
+}
+
+/**
+ * starts the automatic update checking
+ * @param interval milliseconds between interval to check on, defaulkts to 24h
+ */
+export function startUpdateChecking(windowManager: WindowManager, interval = 1000 * 60 * 60 * 24): void {
+  if (isDevelopment) {
+    return;
+  }
+
+  autoUpdater.logger = logger;
+
+  autoUpdater
+    .on("update-available", async (args: UpdateInfo) => {
+      try {
+        const releaseDate = moment(args.releaseDate);
+        const body = `Version ${args.version} was release on ${releaseDate.format("dddd, mmmm dS, yyyy")}.`;
+        windowManager.mainView.webContents.send(NotificationChannelAdd, {
+          args: [{
+            title,
+            body,
+            status: "info",
+            timeout: 5000,
+          }]
+        });
+
+        await autoUpdateCheck(windowManager);
+      } catch (error) {
+        logger.error("[UPDATE CHECKER]: notification failed", { error: String(error) })
+      }
+    })
+    .on("update-not-available", (args: UpdateInfo) => {
+      try {
+        const version = new SemVer(args.version);
+        const stream = version.prerelease === null ? "stable" : "prerelease";
+        const body = `Lens is running the latest ${stream} version.`;
+        windowManager.mainView.webContents.send(NotificationChannelAdd, {
+          args: [{
+            title,
+            body,
+            status: "info",
+            timeout: 5000,
+          }]
+        })
+      } catch (error) {
+        logger.error("[UPDATE CHECKER]: notification failed", { error: String(error) })
+      }
+    })
+
+  async function helper() {
+    while (true) {
+      await autoUpdater.checkForUpdates();
+      await delay(interval);
+    }
   }
+
+  helper()
+    .catch(error => logger.error("[UPDATE CHECKER]: failed with an error", { error: String(error) }));
```

Comment:
So it seems we're interested only in first failure? And if it fail it will stop interval checking?

Reply:
Yes that is the current behaviour, do you think we should try again if it throws an error?

Reply:
I think we should retry on next interval. Is there any reason not to retry?

Reply:
Nope, (and I thought this too haha so it currently has that change)